benchexec/tools/fdse.py

```diff
@@ -0,0 +1,72 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.tools.template
+import benchexec.result as result
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Tool info for FDSE.
+    https://github.com/zbchen/FDSE
+    """
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("fdse")
+
+    def program_files(self, executable):
+        return self._program_files_from_executable(
+            executable, self.REQUIRED_PATHS, parent_dir=True
+        )
+
+    def version(self, executable):
+        return "fdse-testcomp"
+
+    def cmdline(self, executable, options, task, rlimits):
+        new_option = ["--testcomp"]
+        if task.property_file:
+            new_option += [f"--property-file={task.property_file}"]
+        if rlimits.cputime:
+            new_option += [f"-mt={rlimits.cputime}"]
+
+        new_option += [f"-sf={task.single_input_file}"]
+        return [executable] + new_option + options
+
+    def name(self):
+        return "FDSE"
+
+    def determine_result(self, run):
+        """
+        Parse the output of the tool and extract the verification result.
+        This method always needs to be overridden.
+        If the tool gave a result, this method needs to return one of the
+        benchexec.result.RESULT_* strings.
+        Otherwise an arbitrary string can be returned that will be shown to the user
+        and should give some indication of the failure reason
+        (e.g., "CRASH", "OUT_OF_MEMORY", etc.).
+        """
+        for line in run.output:
+            if "[BUG]" in line:
+                if line.find("ASSERTION FAIL!") != -1:
```

Comment:
All the uses of `find()` could be replaced by `str in line`.

Reply:
Oh!  It's my fault, thank you so much.